pkg/lorry/plugin/proto/db_plugin.proto

```diff
@@ -0,0 +1,256 @@
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message LeaveMemberRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica to check the role.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Names of all members.
+  repeated string members = 4;
+
+  // The name of the leave member's Pod.
+  string leave_member = 5;
+
+  // The IP of the leave member's Pod.
+  string leave_member_ip = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message LeaveMemberResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadonlyRequest {
+  DBInfo db_info = 1;
+
+  // The reason for setting the replica as read-only.
+  string Reason = 2;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 3;
+}
+
+message ReadonlyResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message ReadwriteRequest {
+  DBInfo db_info = 1;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 2;
+}
+
+message ReadwriteResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message AccountProvisionRequest {
+  DBInfo db_info = 1;
+
+  // the user name of the new account.
+  string user_name = 2;
+
+  // the password of the new account.
+  string password = 3 [(kb_secret) = true];
+
+  // the role of the new account.
+  string role = 4;
+
+  // Common metadata property for extention
+}
+
+message AccountProvisionResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+message SwitchoverRequest {
+  DBInfo db_info = 1;
+
+  // The pod FQDN of the primary replica.
+  string primary_fqdn = 2;
+
+  // The port on which the primary DB service listens.
+  string primary_port = 3;
+
+  // The Name of the old primary replica to demote.
+  string primary = 4;
+
+  // The Name of the new primary replica.
+  string candidate = 5;
+
+  // Force the switchover to happen even if the primary is not ready.
+  bool force = 6;
+
+  // Common metadata property for extention
+  map<string, string> metadata = 7;
+}
+
+message SwitchoverResponse {
+  // Common metadata property for extention
+  map<string, string> metadata = 1;
+}
+
+// DBPlugin service provides APIs to kubeblocks operator to exec component operation.
+service DBPlugin {
+  rpc GetPluginInfo(GetPluginInfoRequest) returns (GetPluginInfoResponse) {}
+
+  // IsDBReady defines the mechanism to probe the readiness of the database.
+  rpc IsDBReady(IsDBReadyRequest) returns (IsDBReadyResponse) {}
+
+  // GetRole defines the mechanism to probe the role of replicas. The return role
+  // must be one of the names defined in the componentdefinition roles. 
+  rpc GetRole(GetRoleRequest) returns (GetRoleResponse) {}
+
```

Comment:
So postProvisioning and preTerminate won't be handled in the plugin?

Reply:
Currently, those two actions are being handled by Kubernetes Job. once Lorry supports asynchronous execution mode, they will be ported to Lorry.